src/Markdig.Tests/TestPipeTable.cs

```diff
@@ -101,4 +103,72 @@ public void TableWithUnbalancedCodeSpanParsesWithoutDepthLimitError()
         Assert.That(html, Does.Contain("<table"));
         Assert.That(html, Does.Contain("<td>`C</td>"));
     }
+
+    [Test]
+    public void CodeInlineWithPipeDelimitersRemainsCodeInline()
+    {
+        const string markdown = "`|| hidden text ||`";
+
+        var pipeline = new MarkdownPipelineBuilder()
+            .UseAdvancedExtensions()
+            .Build();
+
+        var document = Markdown.Parse(markdown, pipeline);
+
+        var codeInline = document.Descendants().OfType<CodeInline>().SingleOrDefault();
+        Assert.IsNotNull(codeInline);
+        Assert.That(codeInline!.Content, Is.EqualTo("|| hidden text ||"));
+    }
+
+    [Test]
+    public void SingleLineCodeInlineWithPipeDelimitersRendersAsCode()
+    {
+        const string markdown = "`|| hidden text ||`";
+
+        var pipeline = new MarkdownPipelineBuilder()
+            .UseAdvancedExtensions()
+            .Build();
+
+        var html = Markdown.ToHtml(markdown, pipeline);
+
+        Assert.That(html, Is.EqualTo("<p><code>|| hidden text ||</code></p>\n"));
+    }
+
+    [Test]
+    public void MultiLineCodeInlineWithPipeDelimitersRendersAsCode()
+    {
+        const string markdown = """
+`
+|| hidden text ||
+`
+""";
+
+        var pipeline = new MarkdownPipelineBuilder()
+            .UseAdvancedExtensions()
+            .Build();
+
+        var html = Markdown.ToHtml(markdown, pipeline);
+
+        Assert.That(html, Is.EqualTo("<p><code>|| hidden text ||</code></p>\n"));
+    }
+
+    [Test]
+    public void TableCellWithCodeInlineRendersCorrectly()
+    {
+        const string markdown = """
+| Count | A | B | C | D | E |
+|-------|---|---|---|---|---|
+|     0 | B | C | D | E | F |
```

Comment:
Please also add a test case for the case with an indented pipe character in the table.
Something that exercises this added code path:
https://github.com/xoofx/markdig/blob/0e9e80e1cd9c544d77eae59d62efc333bfc0cc3e/src/Markdig/Parsers/Inlines/CodeInlineParser.cs#L88-L91

Reply:
I also removed the while(...) I thought it would be easy to cover but i cannot figure out a way to cover it at all. removing it didn't change the outcome of any test so I deem it safe to remove.